src/solvers/flattening/boolbv_update.cpp

```diff
@@ -74,17 +74,21 @@ void boolbvt::convert_update_rec(
     bvt index_bv=convert_bv(designator.op0());
 
     const array_typet &array_type=to_array_type(type);
-
     const typet &subtype = array_type.subtype();
+    const exprt &size = array_type.size();
 
     std::size_t element_size=boolbv_width(subtype);
 
+    DATA_INVARIANT(
+      size.id() == ID_constant, "update expects constant-sized array");
```

Comment:
Redundant

Reply:
@smowton :-1: - this provides more information that just relying on the cast. But according to guidelines, should be framed in the should form, for clarity:
```
Array should be constant size for boolbv_update
```